controllers/reaper/reaper_telemetry_reconciler.go

```diff
@@ -0,0 +1,101 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reaper
+
+import (
+	"context"
+	"errors"
+
+	k8ssandraapi "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+
+	reaperapi "github.com/k8ssandra/k8ssandra-operator/apis/reaper/v1alpha1"
+
+	"github.com/go-logr/logr"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/telemetry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *ReaperReconciler) reconcileReaperTelemetry(
```

Comment:
This function is never invoked, which results in no service monitor being created.
You need to invoke it in the Reaper controller, like it's done for Stargate in the Stargate controller.
It does seem like we have a code coverage issue as well since this wasn't picked up by the tests.

Reply:
You're right, I totally missed that. I'll get it fixed and I'm working on an e2e test to verify the service monitor bits.

Reply:
Are we planning to install Prometheus for this e2e test? If not, there isn't any benefit in doing an e2e test vs an envtest integration test. Checking the presence of the ServiceMonitor is precisely the type of thing we want to do in envtests.

Reply:
Would you like to maybe just hook into my existing tests over [here](https://github.com/k8ssandra/k8ssandra-operator/tree/main/test/kuttl/test-servicemonitors)? That'll get you everything you need with ~20 lines of additional config.

Reply:
Hooking into existing tests seems like the best thing to do 👍 

@emerkle826 @Miles-Garnsey, my understanding of these tests is that [this kustomization](https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/kuttl/test-servicemonitors/config/single-node-k8ssandra/kustomization.yaml) needs to be modified to add reaper with the telemetry block, and then update [this file](https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/kuttl/test-servicemonitors/03-assert.yaml) to add the assertion for the new service monitor, right?

Reply:
Yes, absolutely correct.